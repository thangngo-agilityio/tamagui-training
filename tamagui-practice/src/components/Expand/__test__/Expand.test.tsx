import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ExpandSidebar from '../index';
import '@testing-library/jest-dom';

describe('ExpandSidebar() ExpandSidebar method', () => {
  it('should match snapshot for Expand', () => {
    const { container } = render(<ExpandSidebar />);

    expect(container).toMatchSnapshot();
  });

  // Happy Path Tests
  describe('Happy Path', () => {
    it('should call the onClick prop when the logout button is clicked', () => {
      const mockOnClick = jest.fn();

      // Render the component with the mock onClick function
      render(<ExpandSidebar onClick={mockOnClick} />);

      // Open the drawer
      const hamburgerButton = screen.getByTestId('button-hamburger');
      fireEvent.click(hamburgerButton);

      // Click the logout button
      const logoutButton = screen.getByTestId('button-logout');
      fireEvent.click(logoutButton);

      // Check if the mock function was called
      expect(mockOnClick).toHaveBeenCalled();
    });
  });

  // Edge Case Tests
  describe('Edge Cases', () => {
    it('should handle the absence of onClick prop gracefully', () => {
      // Render the component without the onClick prop
      render(<ExpandSidebar />);

      // Open the drawer
      const hamburgerButton = screen.getByTestId('button-hamburger');
      fireEvent.click(hamburgerButton);

      // Click the logout button
      const logoutButton = screen.getByTestId('button-logout');
      fireEvent.click(logoutButton);

      // No errors should occur, and the test should pass
    });
  });
});
