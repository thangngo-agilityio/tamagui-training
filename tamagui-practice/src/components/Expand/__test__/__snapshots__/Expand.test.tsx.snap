// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandSidebar() ExpandSidebar method should match snapshot for Expand 1`] = `
<div>
  <div
    class="chakra-stack css-8g8ihq"
  >
    <div
      class="css-9aaew4"
      data-testid="button-hamburger"
    >
      <svg
        fill="none"
        height="21"
        viewBox="0 0 21 21"
        width="21"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 3.1499V5.2499H21V3.1499H0ZM0 9.4499V11.5499H21V9.4499H0ZM0 15.7499V17.8499H21V15.7499H0Z"
          fill="white"
        />
      </svg>
    </div>
  </div>
</div>
`;
